### PR TITLE
Allow #call_or_* to bubble up any result attribute beginning with 'error'

### PR DESCRIPTION
**Pull request by wkotzan-avant**

### CODE REVIEW

**Comment by wkotzan-avant** [comment]
The Travis CI failure is "no environment variables set". Something I did wrong?

**Comment by mikenichols** [comment]
@wkotzan-avant I'm not sure, I haven't seen that error before.

File: lib/actionizer.rb [hunk 1]
```diff
@@ -58,8 +58,11 @@ def method_missing(method_name, *args, &block)
       raise ArgumentError, "#{action_class.name}##{underlying_method}'s result must respond to :failure?"
     end
 
-    error = result.respond_to?(:error) ? result.error : "Unknown: Your result doesn't respond to :error"
-    fail!(error: error) if result.failure?
+    errors = result.to_h.select { |r| r.to_s.start_with? 'error' }
```
Comment (mikenichols) [inline lib/actionizer.rb]: Can you change the block variables to denote it's a hash? `{ |k, _v| k.to_s.start_with? 'error' }`

File: spec/actionizer_spec.rb [hunk 2]
```diff
@@ -175,6 +175,29 @@ def self.call(_params, _arg2)
...
+            include Actionizer
+            def call
+              fail!(errors_xxxxx: 'inner error')
+            end
+          end
+        end
+
+        it 'calls fail! and passes on result.errors_xxxxx and skips calling the second class' do
+          expect(failure_action_class).to receive(:call)
+                                            .with(foo: 'bar').once.and_call_original
```
Comment (mikenichols) [inline spec/actionizer_spec.rb]: I would suggest you just put this on one line. Rubocop should be good with that.

**Comment by wkotzan-avant** [comment]
done

File: spec/actionizer_spec.rb [hunk 3]
```diff
@@ -175,6 +175,29 @@ def self.call(_params, _arg2)
...
+          end
+        end
+
+        it 'calls fail! and passes on result.errors_xxxxx and skips calling the second class' do
+          expect(failure_action_class).to receive(:call)
+            .with(foo: 'bar').once.and_call_original
+          expect(success_action_class).not_to receive(:call)
+
+          result = dummy_class.call(first_class: failure_action_class,
+                                    second_class: success_action_class)
```
Comment (mikenichols) [inline spec/actionizer_spec.rb]: I think you can delete lines 192-195. We're not testing anything that's passed to the action.

**Comment by mikenichols** [comment]
👍

File: lib/actionizer.rb [hunk 4]
```diff
@@ -58,8 +58,9 @@ def method_missing(method_name, *args, &block)
       raise ArgumentError, "#{action_class.name}##{underlying_method}'s result must respond to :failure?"
     end
 
-    error = result.respond_to?(:error) ? result.error : "Unknown: Your result doesn't respond to :error"
-    fail!(error: error) if result.failure?
+    errors = result.to_h.select { |k, _v| k.to_s.start_with? 'error' }
```
Comment (tmertens) [inline lib/actionizer.rb]: doing this on every '*_and_fail' method call is a major drag on performance.

Comment (mikenichols) [inline lib/actionizer.rb]: @tmertens You're totally right. This should only collect the errors hash in case of failure. I'll update it.

**Pull request merged**